internal/app/networkd/pkg/nic/nic.go

```diff
@@ -2,75 +2,323 @@
+	// Need rtnetlink for MTU setting
+	// TODO: possible rtnl enhancement
+	nlConn, err := rtnetlink.Dial(nil)
+	if err != nil {
+		result = multierror.Append(result, err)
+		return nil, result.ErrorOrNil()
+	}
+	iface.rtConn = nlConn
 
 	return iface, result.ErrorOrNil()
 }
+
+// IsIgnored checks the network interface to see if it should be ignored and not configured
+func (n *NetworkInterface) IsIgnored() bool {
+	if n.Ignore || kernel.ProcCmdline().Get(constants.KernelParamNetworkInterfaceIgnore).Contains(n.Name) {
+		return true
+	}
+
+	return false
+}
+
+// Create creates the underlying link if it does not already exist
+func (n *NetworkInterface) Create() error {
+	log.Println("nic.create")
+	iface, err := net.InterfaceByName(n.Name)
+	if err == nil {
+		n.Link = iface
+		return err
+	}
+
+	log.Println("creating link")
+	if err = n.createLink(); err != nil {
+		log.Println("failed to create link")
+		return err
+	}
+
+	log.Println("finding link")
+	iface, err = net.InterfaceByName(n.Name)
+	if err != nil {
+		log.Println("failed to find link")
+		return err
+	}
+	log.Println("found link", iface)
+	n.Link = iface
+
+	return nil
+}
+
+func (n *NetworkInterface) Configure() (err error) {
+	// Ensure link is down if we should ignore the link
+	// so we are at an appropriate default state
+	if n.IsIgnored() {
+		log.Println("ignored", n.Link.Name)
+		return n.rtnlConn.LinkDown(n.Link)
+	}
+
+	// Would be nice to set MTU here, but since we've deferred getting that
+	// information till the Addressing portion, we'll do it there
+
+	if n.Bonded {
+		if err = n.configureBond(n.Link.Index, n.BondSettings); err != nil {
+			return err
+		}
+		// Surely there's a better way to do a *uint32
+		var bondIndex *uint32
+		bondIndex = new(uint32)
+		*bondIndex = uint32(n.Link.Index)
+
+		if err = n.enslaveLink(bondIndex, n.SubInterfaces...); err != nil {
+			return err
+		}
+	}
+
+	log.Println("bringing up", n.Link.Name)
+	if err = n.rtnlConn.LinkUp(n.Link); err != nil {
+		log.Printf("failed to bring up %s: %v", n.Name, err)
+		return err
+	}
+
+	// Wait for link to report up
+	err = retry.Exponential(30*time.Second, retry.WithUnits(250*time.Millisecond), retry.WithJitter(50*time.Millisecond)).Retry(func() error {
+		link, err := n.rtnlConn.LinkByIndex(n.Link.Index)
+		if err != nil {
+			retry.UnexpectedError(err)
+		}
+		if link.Flags&net.FlagUp != net.FlagUp {
+			return retry.ExpectedError(fmt.Errorf("link is not up %s", n.Link.Name))
+		}
+
+		log.Println("link is up", link.Flags.String())
+		return nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to bring up interface%s: %w", n.Link.Name, err)
+	}
+
+	return err
+}
+
+func (n *NetworkInterface) Addressing() error {
+	for _, method := range n.AddressMethod {
+		log.Printf("configuring %s addressing for %s\n", method.Name(), n.Name)
+
+		if err := n.configureInterface(method); err != nil {
+			// Treat as non fatal error when failing to configure an interface
+			log.Println(err)
+			return nil
```

Comment:
We should check for this in the consumer of `Addressing()`, and ignore there instead.